weblate/trans/models/comment.py

```diff
@@ -17,25 +18,39 @@
 from weblate.utils.state import STATE_FUZZY
 
 if TYPE_CHECKING:
-    from weblate.auth.models import AuthenticatedHttpRequest, User
+    from datetime import datetime
+
+    from weblate.auth.models import AuthenticatedHttpRequest, Unit, User
 
 
 class CommentManager(models.Manager):
-    def add(self, request: AuthenticatedHttpRequest, unit, text, scope) -> None:
+    def add(
+        self,
+        request: AuthenticatedHttpRequest,
+        unit: Unit,
+        text: str,
+        scope: str,
+        *,
+        user: User | None = None,
+        timestamp: datetime | None = None,
+    ) -> Comment:
         """Add comment to this unit."""
         # Is this source or target comment?
         unit_scope = unit.source_unit if scope in {"global", "report"} else unit
 
-        user = request.user
```

Comment:
The user variable is also used later in the code, and keeping it `None` value causes crashes (as can be seen in the tests).